QuerySet.update() only updates one document if the field is unique

See [comment](https://github.com/mongodb-labs/django-mongodb/pull/2/files#r1590188926).

A failing test (`target` is a field with `unique=True`):
```python
    def test_multiple_unique(self):
        a_foo = Foo.objects.create(target="aaa")
        b_foo = Foo.objects.create(target="bbb")
        Foo.objects.update(target="ccc")
        a_foo.refresh_from_db()
        b_foo.refresh_from_db()
        self.assertEqual(a_foo.target, "ccc")
        self.assertEqual(b_foo.target, "ccc")

======================================================================
FAIL: test_multiple_unique (update.tests.AdvancedTests.test_multiple_unique)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/django/tests/update/tests.py", line 153, in test_multiple_unique
    self.assertEqual(b_foo.target, "ccc")
AssertionError: 'bbb' != 'ccc'
- bbb
+ ccc
```

--- a/django_mongodb/compiler.py
+++ b/django_mongodb/compiler.py
@@ -244,7 +244,6 @@ class SQLUpdateCompiler(SQLCompiler):
         return self.update(values)
 
     def update(self, values):
-        multi = True
         spec = {}
         for field, value in values:
             if field.primary_key:
@@ -253,14 +252,10 @@ class SQLUpdateCompiler(SQLCompiler):
                 raise NotSupportedError("QuerySet.update() with expression not supported.")
             # .update(foo=123) --> {'$set': {'foo': 123}}
             spec.setdefault("$set", {})[field.column] = value
-
-            if field.unique:
-                multi = False
-
-        return self.execute_update(spec, multi)
+        return self.execute_update(spec)
 
     @wrap_database_errors
-    def execute_update(self, update_spec, multi=True, **kwargs):
+    def execute_update(self, update_spec, **kwargs):
         collection = self.get_collection()
         try:
             criteria = self.build_query().mongo_query
@@ -268,8 +263,7 @@ class SQLUpdateCompiler(SQLCompiler):
             return 0
         options = self.connection.operation_flags.get("update", {})
         options = dict(options, **kwargs)
-        method = "update_many" if multi else "update_one"
-        return getattr(collection, method)(criteria, update_spec, **options).matched_count
+        return collection.update_many(criteria, update_spec, **options).matched_count
 
 
 class SQLAggregateCompiler(SQLCompiler):

